tests/test_add_option_enqueue.py

```diff
@@ -12,7 +12,12 @@
 
 class NotPicklable:
     def __getstate__(self):
-        raise pickle.PicklingError("You shall not serialize me!")
```

Comment:
Thanks for the update, however this is not exactly what I had in mind, sorry for the confusion.

With this change, we simulate a generic `Exception` occurring during `pickle.dumps()`. The `TypeError()` and `pickle.PicklingError()` in the list are just arguments of the `Exception` and are not actually very relevant to the test.

I actually expected two distinct test cases:
* one with `NotPicklable` causing a `pickle.PicklingError()` (the common case) ;
* another test with `NotPicklable` causing a `TypeError()` (the improvement you suggested).

In this way, we cover different possible cases of error that can happen.

However, I understand that you might not be very familiar with the code base. I would suggest that you simply duplicate the original `NotPicklable` error and create a new `NotPicklableTypeError` class (which raises a `TypeError` instead of `pickle.PicklingError`). Then you could also duplicate `test_caught_exception_queue_put()` and `test_not_caught_exception_queue_put()` to test against this new `NotPicklableTypeError()` class.

Duplication is fine, I'll find a way to factorize it eventually, but we can keep it simple for now.